src/cls/rbd/cls_rbd_types.h

```diff
@@ -364,6 +364,34 @@ struct TrashImageSpec {
 };
 WRITE_CLASS_ENCODER(TrashImageSpec);
 
+enum MirrorImageMapState {
```

Comment:
Done. I've taken out the state tracking part and just maintain `instance_id` on disk.